config/config.kafka.extended.hocon

```diff
@@ -0,0 +1,249 @@
+{
+  # Where to read collector payloads from
+  "input": {
+    "type": "Kafka"
+
+    # Name of the Kafka stream to read from
+    "topicName": "collector-payloads"
+
+    # A list of host:port pairs to use for establishing the initial connection to the Kafka cluster
+    # This list should be in the form host1:port1,host2:port2,...
+    "bootstrapServers": "localhost:9092"
+
+    # Optional, Kafka Consumer configuration
+    # See https://kafka.apache.org/documentation/#consumerconfigs for all properties
+    "consumerConf": {
+      "enable.auto.commit": "false"
+      "auto.offset.reset" : "earliest"
+      "session.timeout.ms": "45000"
+      # both key and value deserializers can't be changed
+      "key.deserializer"  : "org.apache.kafka.common.serialization.StringDeserializer"
+      "value.deserializer": "org.apache.kafka.common.serialization.ByteArrayDeserializer"
+    }
+  }
+
+  "output": {
+    # Enriched events output
+    "good": {
+      "type": "Kafka"
+
+      # Name of the Kafka stream to write to
+      "topicName": "enriched"
+
+      # A list of host:port pairs to use for establishing the initial connection to the Kafka cluster
+      # This list should be in the form host1:port1,host2:port2,...
+      "bootstrapServers": "localhost:9092"
+
+      # Optional, Kafka producer configuration
+      # See https://kafka.apache.org/documentation/#producerconfigs for all properties
+      "producerConf": {
```

Comment:
I can imagine wanting to override a setting in my config.  As a user, I would hope this is a legitimate configuration file:

```
  "output": {
    "good": {
      // other fields go here
      "producerConf": {
        "buffer.memory": 1000000
      }
    }
  }
```

But what happens with that config?  Does my map of a single key-value _override_ the `producerConf` map in the `application.conf` or does it _append_ to the map?  If it overrides the map then it's a real problem, because then my serializers are not set properly.

Reply:
it doesn't override, user-provided config is merged with the config from `application.conf` and atm we have a single key-value pair by default (`acks: all`) in `application.conf`

so your config is appended and it is possible to override `acks` setting as well (similar to consumer not allowing to override auto commit setting, we can enforce `acks` as well)

Reply:
I noticed fs2-kafka not setting key serializer/deserializer as expected, I'll send a new commit to enforce key-value serializer/deserializer which will also disable our users to change them

Reply:
Now that you are adding the serializer/deserializer using the `withProperties` method in `Source.scala`... does this mean you can remove them from the `application.conf` file and from the example config file?

Reply:
yes! I'll include remove them from example config file